File: app/src/main/java/com/example/template/datasource/db/SomeDataDAO.java
package com.example.template.datasource.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;


@Dao
public interface SomeDataDAO {

    @Insert
    void add(SomeData someData);

    @Delete
    void delete(SomeData someData);


    @Query("SELECT * FROM SomeData")
    List<SomeData> getAll();


}

File: app/src/main/java/com/example/template/datasource/db/SomeDataDAO.kt
package com.example.template.datasource.db

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface SomeDataDAO {
    @Insert
    fun add(someData: SomeData?)

    @Delete
    fun delete(someData: SomeData?)

    @get:Query("SELECT * FROM SomeData")
    val all: List<SomeData>?
}